### ISSUE TITLE
Apk corrupted when built with Fastlane

### ISSUE THREAD

**Issue by theREMASTERED1** [shr3jn/fastlane-plugin-huawei_appgallery_connect#60]
Hi
Recently we developed an app for Google Play and AppGallery. We have been using this plugin to deploy(Had been working well) to AppGallery but a few versions back after removing Google Play services from the build flavor for Huawei, Our app has been having trouble with processing on AppGallery and seems to get corrupted. The App does not open correctly and will not process on AppGallery.

I then proceeded to build the release APK using the same build command manually and upload it manually to AppGallery which then worked fine and the app was able to process and open correctly

(Note that there are no errors in GitHub actions for the Fastlane deployment, It seems 100% successful)

I've exhausted all options for fixing this so Id like to enquire if it may have something to do with the plugin Fastlane deploy.

The code we have is correct and has not been changed since we started to deploy to AppGallery and did work fine until we we removed Google-services.json from the flavor to make the app available for HMS users. I have been testing using cloud debugging using the working version and there seems to be nothing wrong at all with the application code but when deployed with Fastlane it gets "corrupted"? but doesn't error on the GitHub actions

**Comment by kelokchan** [comment]
facing the same issue as well. The aab upload was stuck on processing in the appgallery. However, if I bundle the aab manually and upload to the appgallery it works fine

**Comment by theREMASTERED1** [comment]
kinda like in the image @kelokchan ?
Good to know I'm not the only one
Also just started recently for you?

Eventually just times out after a good while

<img width="1026" alt="Screenshot 2024-05-13 at 12 28 24" src="https://github.com/shr3jn/fastlane-plugin-huawei_appgallery_connect/assets/103521785/583c761b-8acf-44ee-851d-ee9edfb537c7">

**Comment by theREMASTERED1** [comment]
![fail](https://github.com/shr3jn/fastlane-plugin-huawei_appgallery_connect/assets/103521785/3e1e5374-8a68-4e9f-9a5a-f5f55021bb42)

**Comment by gbrendez** [comment]
had the same experience

**Comment by theREMASTERED1** [comment]
@kelokchan @gbrendez  could you elaborate a bit more on when the issue start to occur and stuff like that if possible?

**Comment by gbrendez** [comment]
> @kelokchan @gbrendez could you elaborate a bit more on when the issue start to occur and stuff like that if possible please?

from my end, it occured randomly. we haven't made any changes to our Fastfile except changing submit_for_review: false because it keeps processing indefinitely

**Comment by theREMASTERED1** [comment]
@gbrendez yep, also just started to occur randomly for me, No Fastlane code was changed, Only removed the google-services from the build

I have been in contact with Huawei developer support and I have linked this issue, Hopefully, we get a fix

**Comment by theREMASTERED1** [comment]
I did get one reply from support on my current ticket but I am still waiting on my ticket directly related to this issue

> There are no Fastlane plugins supported by Huawei. Additionally, the Fastlane plugin you used is incompatible with the new publishing API interface.

@shr3jn plugin might need to updated to account for this change?

**Comment by GeylanKalafMohe** [comment]
I face the same issue and according to huawei support this issue occurs since this package uses an old non maintained API URL to upload the aab. The package need to change to this api: https://developer.huawei.com/consumer/en/doc/AppGallery-connect-References/agcapi-upload-url-new-0000001111685200

@shr3jn  Kindly take a look
```
Dear developer,

About your feedback,

You are invoking the interface to upload the package. However, the interface is an old interface and is no longer maintained. Can you please switch to the new interface according to the following guide.

https://developer.huawei.com/consumer/en/doc/AppGallery-connect-References/agcapi-upload-url-new-0000001111685200

If you have any other related questions, feel free to ask.

Have a nice day.

HUAWEI Developers
```

**Comment by GeylanKalafMohe** [comment]
This method need to be updated

![image](https://github.com/shr3jn/fastlane-plugin-huawei_appgallery_connect/assets/42656134/994f7ba2-9acb-40d1-9a14-348ad9410f20)

**Comment by GeylanKalafMohe** [comment]
temporary "solution" is to run the lane to upload at least the metadata and then uploading the aab manually..

**Comment by shr3jn** [comment]
I'm working on this fix.

**Comment by theREMASTERED1** [comment]
Hy @shr3jn I made a PR with the necessary changes although I'm not too sure if it's correct as I'm not a ruby dev haha

Could you please take a look?

**Comment by shr3jn** [comment]
@theREMASTERED1 no needs changes on the subsequent API as well.

**Comment by theREMASTERED1** [comment]
@shr3jn I may have found a potential spelling mistake that is causing the issue regarding saving the app info for the fileDestUrl

**Comment by shr3jn** [comment]
Fixed on version 1.0.29.

**Comment by theREMASTERED1** [comment]
Thanks alot
